recirq/quantum_chess/swap_updater.py

```diff
@@ -39,6 +39,37 @@ def _satisfies_adjacency(gate: cirq.Operation) -> bool:
     return q1.is_adjacent(q2)
 
 
+def _pairwise_shortest_distances(
+    qubits: Iterable[cirq.GridQubit]
+) -> Dict[Tuple[cirq.GridQubit, cirq.GridQubit], int]:
+    """Precomputes the shortest path length between each pair of qubits.
+
+    Returns:
+        dictionary mapping a pair of qubits to the length of the shortest path
+        between them (disconnected qubits will be absent).
+    """
+    # Do BFS starting from each qubit and collect the shortest path lengths as
+    # we go.
+    # For device graphs where all edges are the same (where each edge can be
+    # treated as unit length) repeated BFS is O(v**2) and is faster than
+    # the Floyd-Warshall algorithm which is O(v**3).
+    # This won't work if in the future we figure out a way to incorporate edge
+    # weights (for example in order to give negative preference to gates with poor
+    # calibration metrics).
```

Comment:
Optional: should we put this comment into the docstring where it is more visibile?
(Or do you think it is so implementation-specific that it belongs in the code?)

Reply:
I think it's worth adding a comment about the runtime complexity to the docstring, but I'd avoid adding more details about the implementation beyond that.